executor/common_kvm_arm64.h

```diff
@@ -331,3 +333,32 @@ static long syz_kvm_vgic_v3_setup(volatile long a0, volatile long a1, volatile l
 	return vgic_fd;
 }
 #endif
+
+#if SYZ_EXECUTOR || __NR_syz_kvm_vgic_its_setup
+
+// Set up the VGICv3 interrupt controller.
+// syz_kvm_vgic_v3_setup(fd fd_kvmvm, ncpus flags[kvm_num_cpus], nirqs flags[kvm_num_irqs])
+static long syz_kvm_vgic_its_setup(volatile long a0)
+{
+	const int vm_fd = a0;
+
+	int vgic_its_fd = kvm_create_device(vm_fd, KVM_DEV_TYPE_ARM_VGIC_ITS);
+	if (vgic_its_fd == -1)
+		return -1;
+
+	int ret = kvm_set_device_attr(vgic_its_fd, KVM_DEV_ARM_VGIC_GRP_CTRL, KVM_DEV_ARM_VGIC_CTRL_INIT, NULL);
+	if (ret == -1) {
+		close(vgic_its_fd);
+		return -1;
+	}
+
+	uint64 gic_its_base_gpa = ARM64_ADDR_GICI_BASE;
+	ret = kvm_set_device_attr(vgic_its_fd, KVM_DEV_ARM_VGIC_GRP_ADDR, KVM_VGIC_ITS_ADDR_TYPE, &gic_its_base_gpa);
```

Comment:
Is there anything here that the fuzzer can't do on its own? All syscalls look simple, normal and straightforward. If it's the case, then it's better to add just a seed for this syscall sequence.
We can't create a pseudo-syscall for each pair/triple of syscalls (sockets will require a hundred). Pseudo-syscalls are intended to work-around gaps in current fuzzer engine capabilities.

Reply:
Fair enough.